src/packs.c

```diff
@@ -167,21 +167,46 @@ int download_subscribed_packs(struct list *subs, struct manifest *mom, bool requ
 	struct sub *sub = NULL;
 	int err;
 	int is_mix = 0;
-	unsigned int list_length = list_len(subs);
-	unsigned int complete = 0;
 	void *download_handle;
+	double download_size;
 
-	info("Downloading packs...\n");
-	download_handle = swupd_curl_parallel_download_start(get_max_xfer(MAX_XFER));
+	download_size = packs_query_total_download_size(subs, mom);
+	if (download_size < 0) {
+		warn("The size of the packs could not be determined, skipping pack download\n");
+		return download_size;
+	}
+	swupd_curl_parallel_download_set_size(download_size);
 
+	download_handle = swupd_curl_parallel_download_start(get_max_xfer(MAX_XFER));
 	swupd_curl_parallel_download_set_callbacks(download_handle, download_successful, download_error, download_free_data);
+
+	info("Downloading packs (%.2lf Mb) for:\n", download_size / 1000000);
+	iter = list_head(subs);
+	while (iter) {
+		sub = iter->data;
+		iter = iter->next;
+
+		if (!is_installed_bundle(sub->component)) {
```

Comment:
This looks like a bug fix that isn't related to this patch. If so, it should be in a different commit.

Reply:
yes, this is a fix to a bug that I found while working on this.... since this fix ended up in a section of code that didn't exist in the original code it would be difficult to separate it in a different commit. If you really think I should try to separate it, it would be easier to move it to a whole different PR.

Reply:
If it's hard you can leave it here. But add a comment in the commit message explaining that you are also fixing this bug.